python/ray/worker.py

```diff
@@ -949,6 +949,10 @@ def _wait_for_and_process_task(self, task):
             task.function_descriptor_list())
         driver_id = task.driver_id()
 
+        # Check if the active driver has changed
+        previous_driver_id = self.task_driver_id
```

Comment:
I don't think this correctly identifies the previous driver ID. I think the Worker class may actually need to store the previous driver ID.

I tried out the PR but it still prints excessive amounts of `Ray driver id: DriverID(7af9149c78f6e8b93cd0c5b49af962cd7b7e9d21)` even when there is only one driver.